src/Nancy.ViewEngines.Razor.BuildProviders/NancyCSharpRazorBuildProvider.cs

```diff
@@ -54,18 +51,31 @@ public override void GenerateCode(AssemblyBuilder assemblyBuilder)
         /// <param name="results">The compilation results for the build provider's virtual path.</param>
         public override Type GetGeneratedType(CompilerResults results)
         {
-            return results.CompiledAssembly.GetType(string.Format(CultureInfo.CurrentCulture, "{0}.{1}", new object[] { this.host.DefaultNamespace, this.host.DefaultClassName }));
+            return results.CompiledAssembly.GetType(string.Format(CultureInfo.CurrentCulture, "{0}.{1}", new object[] { this.Host.DefaultNamespace, this.Host.DefaultClassName }));
+        }
+
+        private RazorEngineHost Host
+        {
+            get
+            {
+                if (this.host == null)
+                {
+                    this.host = new NancyRazorEngineHostFactory().Create(new CSharpRazorCodeLanguage(), this.VirtualPath);

```

Comment:
There's a bit of a hidden dependency here. It might be that this stuff is *never* going to be rewritten by someone in their own implementation of a Nancy app. However, we could move the registrations out to the [RazorViewEngineApplicationStartupRegistrations](https://github.com/NancyFx/Nancy/blob/master/src/Nancy.ViewEngines.Razor/RazorViewEngineApplicationStartupRegistrations.cs) class